components/post_edit_history/edited_post_item/edited_post_item.tsx

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo, useCallback, useState} from 'react';
+
+import {defineMessages, useIntl} from 'react-intl';
+import classNames from 'classnames';
+
+import IconButton from '@mattermost/compass-components/components/icon-button';
+import {CheckIcon} from '@mattermost/compass-icons/components';
+
+import {Post} from '@mattermost/types/posts';
+
+import Constants, {ModalIdentifiers} from 'utils/constants';
+import {imageURLForUser} from 'utils/utils';
+import {t} from 'utils/i18n';
+
+import PostAriaLabelDiv from 'components/post_view/post_aria_label_div';
+import OverlayTrigger from 'components/overlay_trigger';
+import PostBodyAdditionalContent from 'components/post_view/post_body_additional_content';
+import PostMessageContainer from 'components/post_view/post_message_view';
+import Tooltip from 'components/tooltip';
+import Avatar from 'components/widgets/users/avatar';
+import UserProfileComponent from 'components/user_profile';
+import Timestamp, {RelativeRanges} from 'components/timestamp';
+import InfoToast from 'components/info_toast/info_toast';
+
+import RestorePostModal from '../restore_post_modal';
+
+import {PropsFromRedux} from '.';
+
+const DATE_RANGES = [
+    RelativeRanges.TODAY_TITLE_CASE,
+    RelativeRanges.YESTERDAY_TITLE_CASE,
+];
+
+export type Props = PropsFromRedux & {
+    post: Post;
+    isCurrent?: boolean;
+}
+
+const EditedPostItem = ({post, isCurrent = false, postCurrentVersion, actions}: Props) => {
+    const {formatMessage} = useIntl();
+    const [open, setOpen] = useState(isCurrent);
+
+    const openRestorePostModal = useCallback(() => {
+        const restorePostModalData = {
+            modalId: ModalIdentifiers.RESTORE_POST_MODAL,
+            dialogType: RestorePostModal,
+            dialogProps: {
+                post,
+                actions: {
+                    handleRestore,
+                },
+            },
+        };
+
+        actions.openModal(restorePostModalData);
+    }, [actions, post]);
+
+    const togglePost = useCallback(() => {
+        setOpen((prevState) => !prevState);
+    }, []);
+
+    if (!post) {
+        return null;
+    }
+
+    const itemMessages = defineMessages({
+        helpText: {
+            id: t('post_info.edit.restore'),
+            defaultMessage: 'Restore',
+        },
+        currentVersionText: {
+            id: t('post_info.edit.current_version'),
+            defaultMessage: 'Current Version',
+        },
+    });
```

Comment:
Now you can move this const outside of the function component near the `DATE_RANGES` const.  This will help reduce the size of the component function itself.

Reply:
Updated as suggested